src/Servers/Kestrel/Transport.Quic/README.md

```diff
@@ -1,3 +1,167 @@
-For external contributors, msquic.dll isn't available publicly yet. See https://github.com/aspnet/Announcements/issues/393.
+# Getting start with HTTP/3 on Windows and Linux
 
-Credit to Diwakar Mantha and the Kaizala team for the MsQuic interop code.
\ No newline at end of file
+This is a getting started guide for using HTTP/3 on Windows and Linux.
+
+## Prerequisites
+
+1. Download the latest 5.0-preview build of .NET from <https://dotnet.microsoft.com/download/dotnet/5.0.>
+2. Download Edge Canary <https://www.microsoftedgeinsider.com/en-us/download.> This is required to browse websites with HTTP/3.
+
+### Windows
+
+1. Latest [Windows Insider Builds](https://insider.windows.com/en-us/), Insiders Fast build. This is required for Schannel support for QUIC.
+2. Enabling TLS 1.3. Add the following registry keys to enable TLS 1.3.
+
+    ```text
+    reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Server" /v DisabledByDefault /t REG_DWORD /d 1 /f
+    reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Server" /v Enabled /t REG_DWORD /d 1 /f
+    reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client" /v DisabledByDefault /t REG_DWORD /d 1 /f
+    reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client" /v Enabled /t REG_DWORD /d 1 /f
+    ```
+
+### Linux
+
+1. [Building OpenSSL](https://github.com/openssl/openssl/pull/8797) with QUIC support.
+
+    Today, OpenSSL doesn't support QUIC, and currently isn't planing to [support QUIC in the 3.0 release](https://www.openssl.org/blog/blog/2020/02/17/QUIC-and-OpenSSL/). .NET relies on OpenSSL for TLS on Linux. There is an outstanding pull request which adds the required APIs to [support QUIC](https://github.com/openssl/openssl/pull/8797).
+
+    ```text
+    git clone https://github.com/akamai/openssl
+    cd openssl
+    ./config
+    make
+    ```
```

Comment:
Not needed right now, but the best thing is if we can provide a Dockerfile with all this wired up.

Reply:
For sure, much of this can be streamlined in the future. However, I'd rather streamline parts of the product itself (dev certs) before streamlining parts that aren't.